platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ImmutableTransformExecution.java

```diff
@@ -57,17 +55,14 @@ public ImmutableTransformExecution(
     @Override
     public void visitIdentityInputs(InputVisitor visitor) {
         super.visitIdentityInputs(visitor);
-        // This is a performance hack. We could use the regular fingerprint of the input artifact, but that takes longer than
-        // capturing the normalized path and the snapshot of the raw contents, so we are using these to determine the identity
-        FileSystemLocationSnapshot inputArtifactSnapshot = fileSystemAccess.read(inputArtifact.getAbsolutePath());
-        visitor.visitInputProperty(INPUT_ARTIFACT_SNAPSHOT_PROPERTY_NAME, inputArtifactSnapshot::getHash);
+        visitRegularInputs(visitor);
```

Comment:
I don't think this should call `visitRegularInputs`. The inputs should be split between identity inputs and regular inputs.